### ISSUE TITLE
Tensorflow error: expected dense_Dense1_input to have shape [null,3] but got array with shape [3,1]

### ISSUE THREAD

**Issue by frodoe7** [tensorflow/tfjs#6320]
I'm trying to build a TF model which takes a tensor with [1,3] shape

    // example: [1,2,3]

And the output is the same

here's a simple code to reproduce the error

    const {
      tensor1d,
      sequential,
      layers,
    } = require('@tensorflow/tfjs-node');
    
    const model = sequential();

      model.add(
        layers.dense({
          units: 3,
          inputShape: [3],
        })
      );
    
      model.compile({
        loss: 'meanSquaredError',
        optimizer: 'adam',
      });

      let inputTensor = tensor1d([1,2,3]);
      let outputTensor = tensor1d([1,2,3]);

    model.fit(inputTensor, outputTensor);
    
    let result = model.predict(inputTensor); // it fires the error here

The **model.fit** & **model.predict** functions both take the same input shape from **inputTensor** variable, however it fires error in the second function **model.predict**

Any explanation for that error, I'm a bit newbie in Tensorflow
